### ISSUE TITLE
Reduce vertical space in commit graph

### ISSUE THREAD

**Issue by Adarma** [sublimehq/sublime_merge#18]
**Problem description**

For some repos, I am the only developer.
In this case, I don't need to see the author, and I don't really care about the time of commits.
Just the modified file count and commit message on the left, and branch names on the right would be great.

**Preferred solution**

Option to hide the second line, Author and Time from the commit graph and reduce the vertical space.

**Screenshots**

One commit per line, similar to gitx:

![image](https://user-images.githubusercontent.com/4405072/46208666-b4417980-c32b-11e8-93d3-dd2c8e8c5e12.png)

**Comment by Herteby** [comment]
Besides compactness, it also makes it faster to read.

If the view is wide enough, the author/date could be displayed on the right, and they could also be show as a tooltip on hover.

**Comment by tomngo** [comment]
Adding my support for this. Showing more commits by reducing vertical space would produce quantum leaps in benefit. Spotting patterns synoptically is everything.

**Comment by pzac** [comment]
This is the biggest problem I have with SM. GitX commit graph was/is far more efficient and easier on the eyes.

**Comment by chrisjbillington** [comment]
Yes! This and the density of the diffs (#624, #386) are the biggest pain points for me. There is a lot going on in even medium sized projects, and so being able to see more at once is important. Date and commit author being (optionally) hidden unless there is enough horizontal space for them, but shown on a tooltip, would be ideal. I can see literally twice as many commits in tortoisehg:

(this is a simple mostly linear example but our project is yet to convert our more collaborative branchy repos to git - it will be much more important for them!)

![image](https://user-images.githubusercontent.com/1044087/69742988-ad6aee00-110b-11ea-8cf1-e58dd1e5acc0.png)

**Comment by CrazyTim** [comment]
Let me say its currently not bad, but it would be good if we could configure it to be more compact if required. For example, hiding certain elements such as Committer, or Date. 

Some people might like it displayed as a table with columns, but I haven't had a need for that yet.

**Comment by strawlink** [comment]
+1 for being able to hide certain elements.

You can customize the margin and colors in a sublime theme to alleviate the issue. Go to your packages folder and update the theme file (not sure if the file exists by default, mine is located at `AppData\Roaming\Sublime Merge\Packages\User\Merge Dark.sublime-theme` on Windows).

```
{
    "variables":
    {
        "author_label": "rgb(70, 70, 70)",
        "time_label": "rgb(80, 80, 80)",
    },
    "rules":
    [
        {
            "class": "author_label",
            "fg": "var(author_label)"
        },
        {
            "class": "time_label",
            "fg": "var(time_label)"
        },
        {
            "class": "commit_summary_control",
            "content_margin" : [2,1,0,1]
        }
    ]
}
```
![image](https://user-images.githubusercontent.com/6575922/83733500-e30a0a00-a64d-11ea-915f-7f137ce329f9.png)

Full theme documentation is available [here.](https://www.sublimemerge.com/docs/themes)

You _could_ add a negative margin and get it to one-line that way, but unfortunately it also hides the annotations.
```
{
    "class": "commit_summary_control",
    "content_margin" : [2,0,0,-16]
},
```
![image](https://user-images.githubusercontent.com/6575922/83733794-59a70780-a64e-11ea-8463-3f60927a6bf5.png)

**Comment by pzac** [comment]
If you're on a mac you need to add that file to `~/Library/Application Support/Sublime Merge/Packages/User`.
If you use the light theme the file name should be `Merge.sublime-theme`

**Comment by chrisjbillington** [comment]
Thanks so much @strawlink, I can see ~50% more commits now!

![image](https://user-images.githubusercontent.com/1044087/83793785-098a6e00-a66b-11ea-95fa-8a90aa702bcf.png)

This is with the following (font face and colours obviously to taste, but shrinking the annotations was important to increasing the density since they determine the space between the commit message and author/time lines of each row):
```json
{
    "variables":
    {
        "author_label": "rgb(120, 120, 120)",
        "time_label": "rgb(120, 120, 120)",
        "commit_list_bg": "rgb(15, 15, 15)",
        "side_bar_container_bg": "rgb(20, 20, 20)"
    },
    "rules":
    [
        {
            "class": "commit_summary_control",
            "layer0.tint": "rgb(30, 30, 30)"
        },
        {
            "class": "message_label",
            "font.face": "Ubuntu",
            "fg": "white",
            "font.size": 13,
        },
        {
            "class": "author_label",
            "fg": "var(author_label)",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "time_label",
            "fg": "var(time_label)",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "commit_summary_control",
            "content_margin" : [2,3,0,0],
        },
        {
            "class": "tag_annotation",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "branch_annotation",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "remote_annotation",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "head_annotation",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "stash_annotation",
            "font.size": 11,
            "font.face": "Ubuntu"
        },
        {
            "class": "file_annotation",
            "font.size": 11,
            "font.face": "Ubuntu",
            "fg": "rgb(255,255,0)"
        }

    ]
}
```

**Comment by kaala** [comment]
I need a one line layout for commit graph
Is the theme solution only available after buy licenses? seems not work

**Comment by dspasicevs** [comment]
Bump. It would be great if there was toggle setting - which will show comits in one-line fashion. or drop-down in preferences with scale presets from which the user can quickly choose: 25% 50% 75% of normal size

**Comment by Adarma** [comment]
Just saw the new preference `commit_graph_author_date` added in Build 2077.
Was hoping it would close this, but unfortunately it is unrelated :cry:
